### ISSUE TITLE
Laptop Gaze Inference of dataset output

### ISSUE THREAD

**Issue by HU-xiaobai** [CSAILVision/GazeCapture#61]
Hi, now I have a question about the dataset output. The dataset output is two number x and y and it is on phone or iPad, and now I want to reflect to the laptop coordinate. I review your paper and previous question, you said the figure 6 in the paper is about the coordinate for the output. But I could not understand it: in the middle is the camera and iPad has four direction and phone has three direction. Is seems that you rotate the iPad and phone and make the camera as the centre point. 

So I am confused about: 
1. what is the output x and y means?(can you give a example? such as (3.3, 5.5) (although it is said that x and y represent the coordinate of the phone in figure 5) what actually is the centre point(0,0) of the output(I thought it maybe the centre of the phone/iPad before but it seems that it is wrong)? And I do not understand why we should rotate the iPad and phone to get the larger coordinate and make the camera as the centre point, the camera is always on the up bar of the phone and iPad ?
2. Can I directly use the model to predict the output through the webcam of a laptop? should I make a reflection from the coordinate of the phone/iPad to the coordinate of the laptop? could you give a example? for example if the laptop is 60cm x 40 cm and maybe the phone in your dataset is 15cm x 20cm.

I know my question is a little bit long. Thanks for your reading and patience!

**Comment by erkil1452** [comment]
Hi, 
1. (3.3, 5.5) = user gazes 3.3 cm right and 5.5 cm up from the camera lens. We rotate the phones to get a larger screen estate. Just in your example we would never be able to show person a target *above* the camera lens if we did not rotate the device. It is only relevant for data collection, you do not need to worry about any rotations.
2. As long as your webcam is identical to the camera in our iPhone/iPad (some older versions of those) then it should work. You also need to either maintain the viewing distance or account for it. In your example you would only be able to track the gaze in the part of the screen that is covered by the ipad (15x20cm?). If you move away from the screen this scales up linearly (it is the viewing angle that matters at the end).

**Comment by HU-xiaobai** [comment]
Hi, thanks for your answer. I still confused about the question 1. 
1.1 so the camera point is the centre point(0,0) ? And does the camera is on the centre of the up bar of the phone/iPad? 
1.2 let us image a common situation. we have just a single phone or laptop with a camera on its centre of up bar. as you means (3.3, 5.5)  =  user gazes 3.3 cm right and 5.5 cm up from the camera lens, is that means the gaze is out of the phone/laptop scale under this situation? And only when y < 0 the point is on the screen of the phone/laptop? Is it right?

**Comment by erkil1452** [comment]
1) Yes, camera is (0,0) regardless of device or its orientation.
2) Yes, in that case the user would be looking outside of the screen estate. And yes, for any screen point it would be y < 0.

**Comment by HU-xiaobai** [comment]
Hi, so no matter where the camera is, the camera is (0,0) right?(because some laptop cameras are on the down bar of the screen(such as dell laptop)

**Comment by erkil1452** [comment]
Yes, (0,0) is always camera. There is no way to tell where the display is based on the camera image (= network input), so it cannot make any difference.

**Comment by HU-xiaobai** [comment]
OK I get it. Thanks for your answer! really clear

**Comment by ceyxasm** [comment]
Hey @HU-xiaobai I wanted to ask how did you get the model running? 
There are no instructions on the repo itself and I was wondering if you could help.
I have a few questions
1. Did you run Caffe model or Pytorch
2. Is there any way to run it for live inferencing?

**Comment by HU-xiaobai** [comment]
@ceyxasm Hi, it was a long time ago I feel sorry I forgot nearly everything. What I remember is, that we need to collect the training dataset at first and then we inference, it seems we could not live inference if we do not have a training dataset

**Comment by ceyxasm** [comment]
> [@ceyxasm](https://github.com/ceyxasm) Hi, it was a long time ago I feel sorry I forgot nearly everything. What I remember is, that we need to collect the training dataset at first and then we inference, it seems we could not live inference if we do not have a training dataset

@HU-xiaobai  Thanks for your quick reply. Just one last question. Is it worth the effort? Because I am working on gaze detection and tracking and no models/implementations seem to do a decent job. Was this any good. (I was very nice accuracy to tell if a person is looking on screen or off-screen)

**Comment by HU-xiaobai** [comment]
@ceyxasm I am not sure what good degree could make you worth the effort actually. In my memory, in my project, each one needs to be collected 1-3 minute training dataset (depending on the laptop, since my laptop captures 60 pictures a minute.) And during inference, the distance of eye gaze(true label) and the predict label around 1-3cm), but I memorize the model is just CNN, which out of date now. So if you use a more power model, it maybe more power. But if you just detect the person is looking on screen or off-screen, I think it is enough. Lastly, I think it is depond on your purpose, there are two ways: one is collect the person training dataset at the beginning and inference the person later and another is to collect multiple-person training dataset.
